lib/elements/footing.rb

```diff
@@ -0,0 +1,35 @@
+module Elements
+  class Footing
+    attr_reader :concrete
+
+    attr_accessor :step, :dimension1, :dimension2, :height
+
+    def initialize(dimension1:, dimension2:, height:, concrete:, step: 100)
+      @dimension1 = dimension1.to_f
+      @dimension2 = dimension2.to_f
+      @height = height.to_f
+      @step = step.to_f
```

Comment:
what is `step`?

Reply:
I suggest using it to increase (or decrease) each dimension by this step each time we use #incrase_length_x_by_step, for example.

But I think this is functionality for the front-end, and at the core, we modify dimensions based on the front-end requests. And uses them for calculus intentions.

-------------
I didn't modify anything for now.

Reply:
Yeah I agree with that, or at least that doesn't feel like a responsibility for this class. We can think about it later